src/rdkafka_broker.c

```diff
@@ -5238,20 +5267,36 @@ int rd_kafka_brokers_add0(rd_kafka_t *rk, const char *brokerlist) {
                         break;
 
                 rd_kafka_wrlock(rk);
+                if (rk->rk_conf.enable_bootstrap_servers_canonical_resolve) {
```

Comment:
Do you mean `rd_kafka_broker_resolve`? I added it here because `rd_kafka_broker_resolve` is called after the broker is added. By making the changes here, we won't be adding brokers with the wrong addresses initially.

Reply:
There are two places this is being called. One place we use `bootstrap.servers` and other place is external add function. I think we only need to enable this property in `bootstrap.servers` part. Let's add that criteria as well.

Reply:
@emasab - What's your opinion on these both points?